modules/rostests/apitests/shell32/ShellState.cpp

```diff
@@ -0,0 +1,226 @@
+    printf("sizeof(SHELLFLAGSTATE): %d\n", (int)sizeof(SHELLFLAGSTATE));
+    printf("sizeof(CABINETSTATE): %d\n", (int)sizeof(CABINETSTATE));
+
+    pss = &rss.ss;
+    pb = (LPBYTE)pss;
+
+    ret = read_key(&rss);
+    if (ret)
+    {
+        return;
+    }
+
+    dump_pss(pss);
+    ok(rss.dwSize >= 0x24, "rss.dwSize was %ld (0x%lX).\n", rss.dwSize, rss.dwSize);
+
+#define DUMP_LONG(x) printf(#x ": 0x%08X\n", int(x));
+#define DUMP_BOOL(x) printf(#x ": %d\n", !!int(x));
+    DUMP_BOOL(pss->fShowAllObjects);
+    DUMP_BOOL(pss->fShowExtensions);
+    DUMP_BOOL(pss->fNoConfirmRecycle);
+    DUMP_BOOL(pss->fShowSysFiles);
+    DUMP_BOOL(pss->fShowCompColor);
+    DUMP_BOOL(pss->fDoubleClickInWebView);
+    DUMP_BOOL(pss->fDesktopHTML);
+    DUMP_BOOL(pss->fWin95Classic);
+    DUMP_BOOL(pss->fDontPrettyPath);
+    DUMP_BOOL(pss->fShowAttribCol);
+    DUMP_BOOL(pss->fMapNetDrvBtn);
+    DUMP_BOOL(pss->fShowInfoTip);
+    DUMP_BOOL(pss->fHideIcons);
+    DUMP_BOOL(pss->fWebView);
+    DUMP_BOOL(pss->fFilter);
+    DUMP_BOOL(pss->fShowSuperHidden);
+    DUMP_BOOL(pss->fNoNetCrawling);
+    DUMP_LONG(pss->lParamSort);
+    DUMP_LONG(pss->iSortDirection);
+    DUMP_LONG(pss->version);
+    DUMP_LONG(pss->lParamSort);
+    DUMP_LONG(pss->iSortDirection);
+    DUMP_LONG(pss->version);
+    DUMP_BOOL(pss->fSepProcess);
+    DUMP_BOOL(pss->fStartPanelOn);
+    DUMP_BOOL(pss->fShowStartPage);
+#if NTDDI_VERSION >= 0x06000000     // for future use
+    DUMP_BOOL(pss->fIconsOnly);
+    DUMP_BOOL(pss->fShowTypeOverlay);
+    DUMP_BOOL(pss->fShowStatusBar);
+#endif
+
+#define SSF_MASK \
+    (SSF_SHOWALLOBJECTS | SSF_SHOWEXTENSIONS | SSF_NOCONFIRMRECYCLE | \
+     SSF_SHOWCOMPCOLOR | SSF_DOUBLECLICKINWEBVIEW | SSF_DESKTOPHTML | \
+     SSF_WIN95CLASSIC | SSF_DONTPRETTYPATH | SSF_SHOWATTRIBCOL | \
+     SSF_MAPNETDRVBUTTON | SSF_SHOWINFOTIP | SSF_HIDEICONS)
+    // For future:
+    // SSF_AUTOCHECKSELECT, SSF_ICONSONLY, SSF_SHOWTYPEOVERLAY, SSF_SHOWSTATUSBAR
+
+    memset(&FlagState, 0, sizeof(FlagState));
+    SHGetSettings(&FlagState, SSF_MASK);
+
+#define CHECK_FLAG(x) ok(pss->x == FlagState.x, "%s\n", #x)
+    CHECK_FLAG(fShowAllObjects);
+    CHECK_FLAG(fShowExtensions);
+    CHECK_FLAG(fNoConfirmRecycle);
+    //CHECK_FLAG(fShowSysFiles);    // No use
+    CHECK_FLAG(fShowCompColor);
+    CHECK_FLAG(fDoubleClickInWebView);
+    CHECK_FLAG(fDesktopHTML);
+    CHECK_FLAG(fWin95Classic);
+    CHECK_FLAG(fDontPrettyPath);
+    CHECK_FLAG(fShowAttribCol);
+    CHECK_FLAG(fMapNetDrvBtn);
+    CHECK_FLAG(fShowInfoTip);
+    CHECK_FLAG(fHideIcons);
+#if NTDDI_VERSION >= 0x06000000     // for future use
+    CHECK_FLAG(fAutoCheckSelect);
+    CHECK_FLAG(fIconsOnly);
+#endif
+
+#if 1
+    #define DO_IT(x) x
+#else
+    #define DO_IT(x) do { printf(#x ";\n"); x; } while (0)
+#endif
+
+    DO_IT(memset(pss, 0, sizeof(*pss)));
+    DO_IT(pss->dwWin95Unused = 1);
+    ok(pb[4] == 0x01, "%d\n", (pb[4] == 0x01) ? 0 : dump_pss(pss));
```

Comment:
This will print an unnecessary "0". Why not:
```
#define ok_pss(pss, cond) do { ok(cond, "Expected " #cond); if (!(cond)) dump_pss(pss); } while (0)
```
?

Alternatively,
```
ok(pb[4] == 0x01 || dump_pss(pss), "Unexpected pss ^\n");
```
would do the same thing as the current code, without the duplicated condition check and extra "0"

Reply:
OK.